clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -105,7 +105,10 @@ public synchronized Cluster fetch() {
      * will be reset on the next update.
      */
     public synchronized void add(String topic) {
-        topics.put(topic, TOPIC_EXPIRY_NEEDS_UPDATE);
+        if (topics.put(topic, TOPIC_EXPIRY_NEEDS_UPDATE) == null) {
```

Comment:
TBH I'm not sure whether it's a good idea to call `requestUpdate()` on each `.add()` but for now, as it actually does only marking internal variable as update is required, it shouldn't be harmful in any cases. 
ack.